frontend/packages/dev-console/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarParam.tsx

```diff
@@ -3,30 +3,40 @@ import { FormGroup, TextInput } from '@patternfly/react-core';
 import { PipelineResourceTaskParam, PipelineTaskParam } from '../../../../utils/pipeline-augment';
 
 type TaskSidebarParamProps = {
+  hasParamError?: boolean;
   resourceParam: PipelineResourceTaskParam;
   taskParam?: PipelineTaskParam;
   onChange: (newValue: string) => void;
 };
 
 const TaskSidebarParam: React.FC<TaskSidebarParamProps> = (props) => {
-  const { onChange, resourceParam, taskParam } = props;
+  const { hasParamError, onChange, resourceParam, taskParam } = props;
+  const [dirty, setDirty] = React.useState(false);
+
+  const currentValue = taskParam.value?.toString() || '';
+  const emptyIsInvalid = !resourceParam.default;
+
+  const isValid = !(dirty && hasParamError && emptyIsInvalid && currentValue === '');
 
   return (
     <FormGroup
```

Comment:
Why aren't you using the formik fields we have for this?

Reply:
I need to do custom save logic and Formik messing with the value directly caused desync issues with the errors + updating of the values. I figured I'd just manage all things Task related and let Formik manage the form items (name, params, resources on the main page).

I suspect I could find a way to make the logic exist in the yup schema... but I have not yet tackled that.